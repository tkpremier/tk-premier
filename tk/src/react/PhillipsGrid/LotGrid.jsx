import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { LotsContainer } from '../CuratedView/CuratedAuctionContainers';
import FlocklerBanner from '../FlocklerBanner/FlocklerBanner';
import PhillipsLot from '../PhillipsLot/PhillipsLot';


const Lot = (props) => {
  const displayTypeClassName = props.listViewType === 'catalogue'
    ? props.auctionLotDisplayTypeName
    : 'single-cell';
  return (
    <li className={classNames('lot', displayTypeClassName)}>
      <PhillipsLot
        {...props}
        imageTransformation="AuctionLotsView"
        showCuratedView={props.listViewType === 'catalogue'}
        showLotNumber={false}
      />
    </li>
  )
};

const Editorial = props => (
  <li className="col-xs-12"><FlocklerBanner {...props} isBanner /></li>
);


const dict = {
  'Container': LotsContainer,
  'Flockler': Editorial,
  'PhillipsLot': Lot
};

const LotGrid = ({ gridItems, listViewType }) => (
  <ul className="phillips-grid">
    {gridItems.map((item) => {
      const Child = dict[item.componentType];
      return (
        <Child {...item} listViewType={listViewType} />
      );
    })}
  </ul>
);

LotGrid.defaultProps = {
  gridItems: [],
  listViewType: 'grid'
};

LotGrid.propTypes = {
  gridItems: PropTypes.arrayOf(PropTypes.object),
  listViewType: PropTypes.string
};

export default LotGrid;
