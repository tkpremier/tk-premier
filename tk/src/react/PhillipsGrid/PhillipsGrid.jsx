import React, { Component } from 'react';
import PropTypes from 'prop-types';

const GridItem = ({ columns, children, className }) => {
  const { xs, sm, md, lg } = columns;
  return (
    <li className={`phillips-grid-item col-xs-${xs} col-sm-${sm} col-md-${md} col-lg-${lg} ${className}`}>
      {children}
    </li>
  );
};

GridItem.defaultProps = {
  className: '',
  columns: {
    lg: 3,
    md: 4,
    sm: 6,
    xs: 12
  }
};

GridItem.propTypes = {
  className: PropTypes.string,
  columns: PropTypes.shape({
    lg: PropTypes.number,
    md: PropTypes.number,
    sm: PropTypes.number,
    xs: PropTypes.number
  }),
  children: PropTypes.element.isRequired
};

const PhillipsGrid = ({ children, classNames, columns }) => (
  <ul className={`phillips-grid ${classNames}`}>
    {children.map(child => (
      <GridItem
        className={child.props.className}
        columns={columns}
        key={child.key}
      >
        {child}
      </GridItem>
    ))}
  </ul>
);

PhillipsGrid.defaultProps = {
  classNames: '',
  columns: {
    lg: 3,
    md: 4,
    sm: 6,
    xs: 12
  },
  itemsShown: 4,
  end: false
};

PhillipsGrid.propTypes = {
  classNames: PropTypes.string,
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
  columns: PropTypes.shape({
    large: PropTypes.number,
    medium: PropTypes.number,
    small: PropTypes.number
  }),
  maxShown: PropTypes.number,
  concat: PropTypes.bool,
  itemsShown: PropTypes.number,
  end: PropTypes.bool,
};

export default PhillipsGrid;
