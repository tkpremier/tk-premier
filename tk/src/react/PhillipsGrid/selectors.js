import { createSelector } from 'reselect';
import isNull from 'lodash/isNull';
import Container from '../CuratedView/CuratedAuctionContainers';
import filterLots from '../PhillipsFilter/utils/getFilteredLots';
import { parseFilterQuery } from '../utils/filterQueryUtils';


const lotsSelector = state => state.sale.lots;
const editorialsSelector = state => state.sale.flocklerBanners;
const urlQueriesSelector = state => state.urlQueries;
const filterDimensionsSelector = ({ filterData }) => filterData.filterDimensions;
const listViewTypeSelector = ({ listViewType }) => listViewType;

const getCuratedItems = (items) => {
  let container = null;
  return items.reduce((gridItems, item) => {
    if (item.auctionLotDisplayTypeName.indexOf('container') > -1) {
      if (isNull(container) || container.isFilled()) {
        container = new Container({
          auctionLotDisplayTypeName: item.auctionLotDisplayTypeName
        });
      }
      container.addLot(item);
      if (container.isFilled()) {
        gridItems.push(container);
      }
      return gridItems;
    }
    gridItems.push({ ...item, componentType: 'PhillipsLot' });
    return gridItems;
  }, [])
};

const getGridItems = (editorials, filterDimensions, listViewType, lots, urlQueries) => {
  const { sort, filter } = urlQueries;
  const showCurated = listViewType === 'catalogue' && filter.length === 0 && sort.length === 0;
  const filterObject = parseFilterQuery(urlQueries.filter);
  const filteredLots = filterLots(filterObject, filterDimensions, lots).map(lot => ({
    ...lot, componentType: 'PhillipsLot'
  }));
  if (showCurated) {
    return getCuratedItems(lots);
  }
  if (filter.length === 0 && sort.length === 0 && editorials.length > 0) {
    const bannerIndices = [];
    let count = 0;
    const totalItems = filteredLots.length + editorials.length;
    for (let i = 0; i <= totalItems; i++) {
      count++;
      if (count % 6 === 0 && count !== 0) {
        bannerIndices.push(i);
        count = 0;
      }
      if (bannerIndices.length === editorials.length) {
        break;
      }
    }
    editorials.forEach((editorial, i) => {
      const bannerIndex = bannerIndices[i];
      filteredLots.splice(bannerIndex + (i + 1), 0, { ...editorial, componentType: 'Flockler' });
    });
  }

  const [type, dir] = urlQueries.sort.length > 0 ? urlQueries.sort.split('-') : ['', ''];
  return urlQueries.sort.length > 0
    ? filteredLots.sort((a, b) => {
      const desc = dir === 'desc';
      const prop = a.hasOwnProperty(type)
        ? type
        : type === 'maker'
          ? 'makerName'
          : type === 'price'
            ? a['hammerPlusBP'] > 0
              ? 'hammerPlusBP'
              : 'lowEstimate'
            : 'lotNumber';
      const noEstimateA = a.lowEstimate === 0;
      const noEstimateB = b.lowEstimate === 0;
      // no estimate = inquiry only
      if (type === 'price') {
        if (noEstimateA && noEstimateB) {
          return 0;
        }
        if (noEstimateA) {
          return 1;
        }
        if (noEstimateB) {
          return -1;
        }
      }
      if (desc) {
        if (a[prop] < b[prop]) {
          return 1;
        }
        if (a[prop] > b[prop]) {
          return -1;
        }
      } else {
        //asc
        if (a[prop] < b[prop]) {
          return -1;
        }
        if (a[prop] > b[prop]) {
          return 1;
        }
      }
      return 0;
    })
    : filteredLots;
};

const gridItemsSelector = createSelector(
  [
    editorialsSelector,
    filterDimensionsSelector,
    listViewTypeSelector,
    lotsSelector,
    urlQueriesSelector
  ],
  getGridItems
);

export default gridItemsSelector;
