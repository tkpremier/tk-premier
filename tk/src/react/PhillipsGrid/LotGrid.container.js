import { connect } from 'react-redux';
import LotGrid from './LotGrid';
import gridItemsSelector from './selectors';

const mapStateToProps = state => ({
  gridItems: gridItemsSelector(state),
  listViewType: state.listViewType
});

export default connect(mapStateToProps)(LotGrid);
