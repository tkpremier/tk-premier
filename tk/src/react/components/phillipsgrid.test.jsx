import React from 'react';
import EditorialItem from './editorialitem';
import PhillipsGrid from './phillipsgrid';
import renderer from 'react-test-renderer';

const items=[{
   htmlCaption:"Feature Video | Photographs",
   imageUrl:"https://8675fa3b12fbaae97df1-49af760ec3cce3b92480c70a3569b570.ssl.cf2.rackcdn.com/HomePage/featuredContent-186.jpg",
   url:"https://www.phillips.com/Video/default?VideoID=190"
   },
   {
   htmlCaption:"Feature Video | Photographs",
   imageUrl:"https://8675fa3b12fbaae97df1-49af760ec3cce3b92480c70a3569b570.ssl.cf2.rackcdn.com/HomePage/featuredContent-186.jpg",
   url:"https://www.phillips.com/Video/default?VideoID=190"
   },
   {
   htmlCaption:"Feature Video | Photographs",
   imageUrl:"https://8675fa3b12fbaae97df1-49af760ec3cce3b92480c70a3569b570.ssl.cf2.rackcdn.com/HomePage/featuredContent-186.jpg",
   url:"https://www.phillips.com/Video/default?VideoID=190"
   },
   {
   htmlCaption:"Feature Video | Photographs",
   imageUrl:"https://8675fa3b12fbaae97df1-49af760ec3cce3b92480c70a3569b570.ssl.cf2.rackcdn.com/HomePage/featuredContent-186.jpg",
   url:"https://www.phillips.com/Video/default?VideoID=190"
   }
   ]

it('should render a grid', () => {
    const grid = renderer.create(
        <PhillipsGrid
            listClass="featuredContent"
            itemClass="grid-item"
            header="Features"
            numberOfItemsShown="4"
        >
         {items.map(props=> <EditorialItem key={Math.random()} {...props}/>)}

       </PhillipsGrid>
    );
    const tree = grid.toJSON();
    expect(tree).toMatchSnapshot();
});
