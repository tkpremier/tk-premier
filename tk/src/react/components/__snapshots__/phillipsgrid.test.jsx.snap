// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a grid 1`] = `
<div
  className="phillips-grid col-xs-12"
>
  <h2>
    Features
  </h2>
  <ul
    className="featuredContent row"
  >
    <li
      className="undefined grid-item"
    >
      <a
        href="https://www.phillips.com/Video/default?VideoID=190"
      >
        <div
          className="content-wrapper"
        >
          <div
            className="image-container"
            style={
              Object {
                "backgroundImage": "url(https://8675fa3b12fbaae97df1-49af760ec3cce3b92480c70a3569b570.ssl.cf2.rackcdn.com/HomePage/featuredContent-186.jpg)",
                "backgroundPosition": "50% 50%",
                "backgroundSize": "cover",
              }
            }
          />
          <span
            className="caption"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Feature Video | Photographs",
              }
            }
          />
        </div>
      </a>
    </li>
    <li
      className="undefined grid-item"
    >
      <a
        href="https://www.phillips.com/Video/default?VideoID=190"
      >
        <div
          className="content-wrapper"
        >
          <div
            className="image-container"
            style={
              Object {
                "backgroundImage": "url(https://8675fa3b12fbaae97df1-49af760ec3cce3b92480c70a3569b570.ssl.cf2.rackcdn.com/HomePage/featuredContent-186.jpg)",
                "backgroundPosition": "50% 50%",
                "backgroundSize": "cover",
              }
            }
          />
          <span
            className="caption"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Feature Video | Photographs",
              }
            }
          />
        </div>
      </a>
    </li>
    <li
      className="undefined grid-item"
    >
      <a
        href="https://www.phillips.com/Video/default?VideoID=190"
      >
        <div
          className="content-wrapper"
        >
          <div
            className="image-container"
            style={
              Object {
                "backgroundImage": "url(https://8675fa3b12fbaae97df1-49af760ec3cce3b92480c70a3569b570.ssl.cf2.rackcdn.com/HomePage/featuredContent-186.jpg)",
                "backgroundPosition": "50% 50%",
                "backgroundSize": "cover",
              }
            }
          />
          <span
            className="caption"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Feature Video | Photographs",
              }
            }
          />
        </div>
      </a>
    </li>
    <li
      className="undefined grid-item"
    >
      <a
        href="https://www.phillips.com/Video/default?VideoID=190"
      >
        <div
          className="content-wrapper"
        >
          <div
            className="image-container"
            style={
              Object {
                "backgroundImage": "url(https://8675fa3b12fbaae97df1-49af760ec3cce3b92480c70a3569b570.ssl.cf2.rackcdn.com/HomePage/featuredContent-186.jpg)",
                "backgroundPosition": "50% 50%",
                "backgroundSize": "cover",
              }
            }
          />
          <span
            className="caption"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Feature Video | Photographs",
              }
            }
          />
        </div>
      </a>
    </li>
  </ul>
  <div
    className="show-more"
    onClick={[Function]}
  >
    Explore More
  </div>
</div>
`;
